components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/services/impl/AutoscalerServiceImpl.java

```diff
@@ -337,6 +337,28 @@ private void addApplicationSignUp(ApplicationContext applicationContext, String
         }
     }
 
+    private void removeApplicationSignUp(ApplicationContext applicationContext){
+        try {
+            if (log.isInfoEnabled()) {
+                log.info(String.format("Removing application signup: [application-id] %s",
+                        applicationContext.getApplicationId()));
+            }
+
+            StratosManagerServiceClient serviceClient = StratosManagerServiceClient.getInstance();
+
+            ApplicationSignUp applicationSignUp[] = serviceClient.getApplicationSignUps(applicationContext.getApplicationId());
+            for(ApplicationSignUp appSignUp : applicationSignUp)
+            {
+                serviceClient.removeApplicationSignUp(appSignUp.getApplicationId(), appSignUp.getTenantId());
```

Comment:
Since applicationSignUps is an array there could be null elements inside the array. Therefore better to check null for the element before invoking serviceClient.removeApplicationSignUp() method.
Thanks